scanner/scanner.go

```diff
@@ -202,11 +231,168 @@ func scanNot(s *Scanner) (token.Token, []byte) {
 }
 
 func scanDoubleQuote(s *Scanner) (token.Token, []byte) {
+	t := token.String
+	next, rOffset := scanInDoubleQoutes(s)
+	switch next {
+	case '@', '$', '{':
+		t = token.StringPart
+		s.pushCtx(stateInDoubleQoutes)
+	case '"':
+		s.next()
+	}
+	off := s.offset - rOffset
+	if off > s.begin+1 {
+		off--
+	}
+	return t, s.src[s.begin+1 : off]
+}
+
+func scanInDoubleQoutes(s *Scanner) (byte, int) {
```

Comment:
__[gometalinter]__
* [gocyclo] cyclomatic complexity 17 of function scanInDoubleQoutes() is high (> 10)

<!-- Comment by [SideCI](https://sideci.com) -->